contracts/pools/PoolCollection.sol

```diff
@@ -970,7 +1027,8 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
 
         // ensure that the average rate is reset when the pool is being emptied
         if (amounts.newBaseTokenTradingLiquidity == 0) {
-            data.averageRate.rate = zeroFraction112();
+            data.averageRates.rate = zeroFraction112();
+            data.averageRates.invRate = zeroFraction112();
```

Comment:
At least mathematically, this initialization (occurring in a few other places) makes no sense, because the inverse of 0 is infinity. I am not entirely convinced that this is just a philosophical debate
We should probably think of a different initialization.
One option is to let both rates represent zero liquidity (i.e., set each rate to the invalid fraction 0:0).
Another option is to let each rate represent a liquidity of 1 BNT and 1 TKN.
So for example, if TKN is USDC, then:
- Set `rate = {n: 1000000000000000000, d: 1000000}`
- Set `rate = {n: 1000000, d: 1000000000000000000}`

The problem with the second option is that we need to rely on `decimals`.

Reply:
This is only an initial value and is being overridden after `enableTrading`, so this comment is irrelevant.